python_submitty_utils/submitty_utils/submitty_schema_validator.py

```diff
@@ -0,0 +1,186 @@
+"""For use in validating submitty configuration objects against json schemas."""
+
+
+import json
+from jsonschema import validate
+import jsonref
+
+
+class SubmittySchemaException(Exception):
+    """An exception capable of printing helpful information about schema errors."""
+
+    def __init__(self, config_json, schema, message, title, schema_error):
+        """Use to aid in printing helpful information about schema errors."""
+        super(SubmittySchemaException, self).__init__(schema_error.message)
+        self.config_json = config_json
+        self.schema = schema
+        self.my_message = message
+        self.title = title
+        self.schema_message = schema_error.message
+
+    def print_human_readable_error(self):
+        """Use to print a human readable version of this exception."""
+        print()
+        print('{0}:'.format(self.my_message))
+        if self.schema_message is not None:
+            print(self.schema_message)
+            print(("The portion of your configuration that caused "
+                   "the error is:"))
+            print(json.dumps(self.config_json, indent=4))
```

Comment:
This should probably be printed to stderr

Reply:
Fixed in upcoming commit.